CMakeLists.txt

```diff
@@ -207,6 +207,7 @@ include(TryAppendCFlags)
 if(MSVC)
   # Keep the following commands ordered lexicographically.
   try_append_c_flags(/W2) # Moderate warning level.
+  try_append_c_flags(/w24334) # Enable warning C4334 "result of 32-bit shift implicitly converted to 64 bits" as a level 2 one.
```

Comment:
I don't think we should try to enable more warnings here manually.

 We may want to go with /W3  (with some explicitly disabled) but this could be part of another PR then, though I'm not entirely if I would to support that change. MSVC is very picky about some things, and we may run into other false positives in the future, requiring us to make more changes etc.

Reply:
[Removed](https://github.com/bitcoin-core/secp256k1/pull/1292#issuecomment-1531403512).